sparrow-py/pysrc/sparrow_py/_timestream.py

```diff
@@ -702,7 +702,8 @@ def lag(self, n: int) -> Timestream:
         Timestream
             Timestream containing the value `n` points before each point.
         """
-        return Timestream._call("lag", n, self)
+        # hack to support structs/lists (as collect supports lists)
+        return self.collect(max=n + 1, min=n + 1, window=None)[0]
```

Comment:
You shouldn't need to specify `window=None` (unless you want to) since it has a default.